cellengine/resources/gate.py

```diff
@@ -1,124 +1,110 @@
+        try:
+            return func(*args, **kwargs)
+        except Exception as err:
+            if type(err) is ValueError:
+                raise err
+            raise RuntimeError(f"Incorrect arguments passed to {func.__name__}.")
+
+    return inner_function
+
+
+def deep_update(source, overrides):
+    """
+    Update a nested dictionary or similar mapping.
+    Modify ``source`` in place.
+    """
+    for key, value in overrides.items():
+        if isinstance(value, Mapping) and value:
+            returned = deep_update(source.get(key, {}), value)
+            source[key] = returned
+        else:
+            source[key] = overrides[key]
+    return source
+
+
+@define(repr=False)
+class Gate:
+    _id: str = field(on_setattr=readonly)
+    experiment_id: str = field(on_setattr=readonly)
+    gid: str = field(on_setattr=readonly)
+    type: str
+    x_channel: str
+    model: Dict
+    tailored_per_file: bool = False
+    fcs_file_id: Optional[str] = None
+    names: Optional[List[str]] = field(default=None)
+    name: Optional[str] = field(default=None)
+    y_channel: Optional[str] = field(default=None)
+
+    def __repr__(self):
+        if self.name:
+            label, name = ("name", self.name)
         else:
-            return cls._build_gate(gates)
+            label, name = ("names", str(self.names))
+        return f"{self.type}(_id='{self._id}', {label}='{name}')"
 
-    @classmethod
-    def _build_gate(cls, gate):
-        """Get the gate type and return instance of the correct subclass."""
-        module = importlib.import_module(__name__)
-        gate, _ = cls._separate_gate_and_population(gate)
-        gate_type = getattr(module, gate["type"])
-        return gate_type(properties=gate)
+    @staticmethod
+    def _format_gate(gate, **kwargs):
+        module = importlib.import_module("cellengine")
+        _class = getattr(module, gate["type"])
+        return _class._format(**gate)
 
     @classmethod
-    def _separate_gate_and_population(cls, gate):
-        try:
-            if "gate" in gate.keys():
-                return gate["gate"], [k for k in gate.keys() if k != "gate"]
-            else:
-                return gate, None
-        except KeyError:
-            raise ValueError("Gate payload format is invalid")
+    def create_many(
+        cls,
+        gates: List[Dict],
+    ):
+        experiment_id = set([g["experiment_id"] for g in gates])
+        if len(experiment_id) != 1:
+            raise RuntimeError("Created gates must all be in the same Experiment.")
 
-    @staticmethod
-    def delete_gates(
-        experiment_id, _id: str = None, gid: str = None, exclude: str = None
-    ) -> None:
-        """Deletes a gate or a tailored gate family.
+        formatted_gates = [cls._format_gate(gate) for gate in gates]
+        return ce.APIClient().create(
+            [Gate(id=None, **g) for g in formatted_gates], create_population=False
+        )  # type: ignore
```

Comment:
Can this be resolved?

Reply:
The issue is actually the `Gate(*args)` call; because there's no explicit __init__ fn mypy/pyright gets mad about this.

I _did_ fix the typing error in `APIClient().create()`, however